napari/layers/shapes/_tests/test_shapes_mouse_bindings.py

```diff
@@ -76,31 +61,33 @@ def test_not_adding_or_selecting_shape(create_known_shapes_layer, Event):
 
 
 @pytest.mark.parametrize('shape_type', ['rectangle', 'ellipse', 'line'])
-def test_add_simple_shape(shape_type, create_known_shapes_layer, Event):
+def test_add_simple_shape(shape_type, create_known_shapes_layer):
     """Add simple shape by clicking in add mode."""
     layer, n_shapes, known_non_shape = create_known_shapes_layer
 
     # Add shape at location where non exists
     layer.mode = 'add_' + shape_type
-    layer.position = known_non_shape
+    position = known_non_shape
 
     # Simulate click
     event = ReadOnlyWrapper(
-        Event(type='mouse_press', is_dragging=False, modifiers=[])
+        CursorEvent(data_position=position, type='mouse_press')
     )
     mouse_press_callbacks(layer, event)
 
     known_non_shape_end = [40, 60]
-    layer.position = known_non_shape_end
+    position = known_non_shape_end
     # Simulate drag end
     event = ReadOnlyWrapper(
-        Event(type='mouse_move', is_dragging=True, modifiers=[])
+        CursorEvent(
+            data_position=position, type='mouse_move', is_dragging=True
+        )
```

Comment:
brain dump:

- is there any reason to not have "is_dragging" be bool but the original click even where the drag started as a value ? Would this be of any help ?

Reply:
It's an interesting idea, in the vispy event the dragging event will actually contain the full history of mouse events during the drag, which i could imagine someone using. it might be nice, i can look into a couple options before deciding